src/time.c

```diff
@@ -145,3 +145,19 @@ _dispatch_timeout(dispatch_time_t when)
 	now = _dispatch_absolute_time();
 	return now >= when ? 0 : _dispatch_time_mach2nano(when - now);
 }
+
+uint64_t
+_dispatch_time_to_epoch_time(dispatch_time_t when)
+{
+	if (when == DISPATCH_TIME_FOREVER) {
+		return DISPATCH_TIME_FOREVER;
+	}
+	if (when == 0) {
```

Comment:
this is wrong too, I removed it afterwards: you want to return _dispatch_get_nanoseconsd() for that one (or fallback through the last return that will do the right thing.